File: src/main/java/Piruru/cards/InfiniteScalingShit.java
package Piruru.cards;

import Piruru.abstracts.PiruruCard;
import Piruru.actions.RecoverAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;



public class InfiniteScalingShit extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(InfiniteScalingShit.class.getSimpleName()));
    private static final int COST = 0;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.UNCOMMON;
    private static final CardTarget TARGET = CardTarget.SELF;
    private static final int RECOVER_AMT = 1;
    private static final int RECOVER_UP = 2;

    public InfiniteScalingShit() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, RECOVER_UP, COST);
        baseMagicNumber = magicNumber = RECOVER_AMT;
        exhaust = true;
    }

    @Override
    public void upgrade() {
        timesUpgraded++;
        if (!upgraded) {
            rawDescription = cardStrings.UPGRADE_DESCRIPTION;
            initializeDescription();
        }
        upgraded = true;
        name = cardStrings.NAME + "+" + timesUpgraded;
        initializeTitle();
        upgradeMagicNumber(RECOVER_UP);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new RecoverAction(magicNumber));
    }
}

File: src/main/kotlin/Cirno/cards/InfiniteScalingShit.kt
package Cirno.cards

import Cirno.Cirno.Statics.makeID
import Cirno.abstracts.PiruruCard
import Cirno.actions.RecoverAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.monsters.AbstractMonster


class InfiniteScalingShit : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, RECOVER_UP, COST) {
    init {
        magicNumber = RECOVER_AMT
        baseMagicNumber = magicNumber
        exhaust = true
    }

    override fun upgrade() {
        timesUpgraded++
        if (!upgraded) {
            rawDescription = cardStrings.UPGRADE_DESCRIPTION
            initializeDescription()
        }
        upgraded = true
        name = cardStrings.NAME + "+" + timesUpgraded
        initializeTitle()
        upgradeMagicNumber(RECOVER_UP)
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        act(RecoverAction(magicNumber))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(InfiniteScalingShit::class.java.simpleName))
        private val COST = 0
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.UNCOMMON
        private val TARGET = AbstractCard.CardTarget.SELF
        private val RECOVER_AMT = 1
        private val RECOVER_UP = 2
    }
}
